apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -60,248 +73,192 @@ export function OAuth2Form() {
-            if (!accessCode || !accessCode.access_code) {
-              onError(
-                new Error(
-                  'Something went wrong - the access code is missing from the Synapse call.',
-                ),
-              )
-              return
-            }
-            // done!  redirect with access code.
-            const redirectUri = getURLParam('redirect_uri')
-            setRedirectURL(
-              `${redirectUri}?${getStateParam()}code=${encodeURIComponent(
-                accessCode.access_code,
-              )}`,
-            )
-          }
-        })
-        .catch(_err => {
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-          onError(_err)
-        })
+    const authRequest: OIDCAuthorizationRequest = {
+      clientId: clientId,
+      scope: scope,
+      responseType: responseType,
+      redirectUri: redirectUri,
+      nonce: queryParams.get('nonce') || undefined,
     }
-    return () => {
-      isSubscribed = false
+    const claimsString = queryParams.get('claims')
+    if (claimsString) {
+      authRequest.claims = JSON.parse(claimsString)
     }
-  }, [accessToken, isConsenting, onError])
+    return authRequest
+  }, [queryParams])
 
-  const onGoBack = () => {
-    window.history.back()
-  }
+  const { data: hasUserAuthorizedOAuthClient } =
+    SynapseQueries.useGetHasUserAuthorizedOAuthClient(
+      oidcAuthorizationRequestFromSearchParams!,
+      {
+        enabled: Boolean(oidcAuthorizationRequestFromSearchParams),
+      },
+    )
 
-  const onDeny = () => {
-    let redirect: string
-    sendGTagEvent('UserDeniedConsent')
-    if (oauthClientInfo && oauthClientInfo.client_uri) {
-      redirect = oauthClientInfo.client_uri
-    } else {
-      redirect = getURLParam('redirect_uri')!
+  const { mutate: consentToRequest } = SynapseQueries.useConsentToOAuth2Request(
+    {
+      onSuccess: accessCode => {
+        if (!accessCode || !accessCode.access_code) {
+          onError(
+            new Error(
+              'Something went wrong - the access code is missing from the Synapse call.',
+            ),
+          )
+          return
+        }
+        // done!  redirect with access code.
+        const redirectUri = queryParams.get('redirect_uri')!
+        redirectToURL(
+          `${redirectUri}?${getStateParam()}code=${encodeURIComponent(
+            accessCode.access_code,
+          )}`,
+        )
+      },
+      onError: e => {
+        onError(e)
+      },
+    },
+  )
+
+  const onConsent = useCallback(() => {
+    sendGTagEvent('UserConsented')
+    if (oidcAuthorizationRequestFromSearchParams) {
+      consentToRequest(oidcAuthorizationRequestFromSearchParams)
     }
-    setRedirectURL(redirect)
-  }
+  }, [consentToRequest, oidcAuthorizationRequestFromSearchParams])
 
-  // if user has already consented to this client request, no need to ask again
+  // Handle auto-consent when the user has already consented
   useEffect(() => {
```

Comment:
The body of this `useEffect` changed, but its purpose is the same in this diff--handling the case where the user has already consented to the scope/claims requested by the client.